src/osip-opcua-test/src/test/java/edu/kit/pse/osip/core/opcua/TankTest.java

```diff
@@ -0,0 +1,82 @@
+package edu.kit.pse.osip.core.opcua;
+
+import edu.kit.pse.osip.core.io.networking.RemoteMachine;
+import edu.kit.pse.osip.monitoring.controller.TankClient;
+import edu.kit.pse.osip.simulation.controller.TankServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests the tank servers and clients
+ * @author Hans-Peter Lehmann
+ * @version 1.0
+ */
+public class TankTest {
+    private static final int PORT = 12686;
+    private static final int INTERVAL = 500;
+    private TankServer server;
+    private TankClient client;
+
+    /**
+     * Initializes server and client
+     * @throws Exception If something goes wrong
+     */
+    @Before
+    public void setup() throws Exception {
+        server = new TankServer(PORT);
+        client = new TankClient(new RemoteMachine("127.0.0.1", PORT));
+        server.start();
+        client.connectClient();
+    }
+
+    /**
+     * Closes server and client
+     * @throws Exception If something goes wrong
+     */
+    @After
+    public void cleanup() throws Exception {
+        client.connectClient();
```

Comment:
Ich habe jetzt nicht mehr genau in den Code geschaut. Aber kann es sein, dass connectClient() hier die falsche Methode ist?